django/db/models/fields/composite.py

```diff
@@ -0,0 +1,146 @@
+from django.core.exceptions import FieldDoesNotExist
+from django.db.models import Field
+from django.db.models.fields.related_lookups import MultiColSource
+from django.db.models.lookups import Exact, In
+from django.db.models.signals import class_prepared
+from django.utils.functional import cached_property
+
+
+class TupleExact(Exact):
```

Comment:
I wonder if there might be a way to define a `TupleLookupMixin` that can be mixed with `Exact`, `In`, `GreaterThan`, and others to do the right thing by default.

Reply:
I could remove the `TupleExact` class and handle the case of comparing tuples inside `Exact`, if that's what you're suggesting? It would be a bit more elegant I suppose.

Reply:
So I looked into it and I started implementing the other lookups. I don't think it's possible to define a mixin that does the right thing by default, but it's possible to make some abstractions to reduce duplication.

Reply:
`gt`, `gte`, `lt`, `lte` have been implemented. Also, `TupleLookupMixin` has been added. Please look into it when you have some time. It's not exactly what you wanted, there's no default behavior as each lookup has their own complexity, but I think this is the only way to implement this.